addons/hr_holidays/models/hr_leave.py

```diff
@@ -1282,10 +1282,13 @@ def _check_approval_update(self, state):
                     if holiday.employee_id == current_employee:
                         raise UserError(_('Only a Time Off Manager can approve/refuse its own requests.'))
 
-                    if (state == 'validate1' and val_type == 'both') or (state == 'validate' and val_type == 'manager') and holiday.holiday_type == 'employee':
+                    if (state == 'validate1' and val_type == 'both') and holiday.holiday_type == 'employee':
                         if not is_officer and self.env.user != holiday.employee_id.leave_manager_id:
                             raise UserError(_('You must be either %s\'s manager or Time off Manager to approve this leave') % (holiday.employee_id.name))
 
+                    if (state == 'validate' and val_type == 'manager') and self.env.user != holiday.employee_id.leave_manager_id:
+                        raise UserError(_('You must but %s\'s Manager to approve this leave') % (holiday.employee_id.name))
```

Comment:
```suggestion
                        raise UserError(_('You must be %s\'s Manager to approve this leave', holiday.employee_id.name))
```

Reply:
I took the line from above, should I also correct it ? :D

Reply:
I'd say no, as you are are not directly changing it :-D